### PR TITLE
When running `antler-proj` without arguments help will be displayed

### LINKED ISSUE #20: Display help when running antler-proj without arguments
It doesn't describe option which allows to declare a name of the app, lib or dep for manipulations.
And addition options needed to specific commands like options for declaration of a programming language, build options an so on.

Actually the CLI syntax has two layers of commands and --help option (as other options) is context specific, i.e. prints a help text dependent on the command and subcommand but it is not described anywhere.
IMHO this principle should be described in the help text which prints antler-proj after run without options.

root@c1742fb0fdb3:/app# antler-proj add  --help
Add an app, dependency, library or test to your project.
Usage: antler-proj add [OPTIONS] [path] [SUBCOMMAND]

Positionals:
  path TEXT                   This is the root path to create the project in.

Options:
  -h,--help                   Print this help message and exit
  -p TEXT                     This is the root path to create the project in.

Subcommands:
  app                         Add a new app to your project.
  lib                         Add a new library to your project.
  dep                         Add a new dependency to the project.
root@c1742fb0fdb3:/app# antler-proj update --help
Update an app, dependency, library or test to your project.
Usage: antler-proj update [OPTIONS] [path] [SUBCOMMAND]

Positionals:
  path TEXT                   This must be the path to the `project.yml` or the path containing it.

Options:
  -h,--help                   Print this help message and exit
  -p TEXT                     This must be the path to the `project.yml` or the path containing it.

Subcommands:
  app                         Remove app from the project.
  lib                         Remove lib from the project.
  dep                         Remove a dependency from the project.
root@c1742fb0fdb3:/app# antler-proj remove --help
Add an app, dependency, library or test to your project.
Usage: antler-proj remove [OPTIONS] path [SUBCOMMAND]

Positionals:
  path TEXT REQUIRED          This must be the path to the `project.yml` or the path containing it.

Options:
  -h,--help                   Print this help message and exit
  -p TEXT REQUIRED            This must be the path to the `project.yml` or the path containing it.

Subcommands:
  app                         Remove app from the project.
  lib                         Remove lib from the project.
  dep                         Remove a dependency from the project.

### PR DESCRIPTION
**Pull request by mikelik**
Resolves #20 

Before:
```
$ antler-proj
Please run one of the subcommands available. Use --help to see what is available.
```

After:
```
$ ./antler-proj 
Antelope Smart Contract Project Management Tool
Usage: antler-proj [OPTIONS] [SUBCOMMAND]

Options:
  -h,--help                   Print this help message and exit
  -V,--version                get the version of antler-proj

Subcommands:
  validate                    Validate a project.
  update                      Update an app, dependency, library or test to your project.
  remove                      Add an app, dependency, library or test to your project.
  populate                    Populate a project's dependencies and CMake.
  init                        Initialize a new project creating the directory tree and a `project.yaml` file.
  build                       Build a project.
  add                         Add an app, dependency, library or test to your project.

Please run one of the subcommands with --help option to get detailed help. Example: antler-proj init --help
```

### CODE REVIEW

**Review by jolly-fellow — COMMENTED** [review]
IMHO this line is important and should be added to the bottom as shown in the comment to the issue.

```
Please run one of the subcommands with --help option to get detailed help for the subcommand. Example: ./antler-proj add --help
```

**Review by jolly-fellow — APPROVED** [review]

File: tools/main.cpp [hunk 1]
```diff
@@ -26,13 +26,15 @@ static V depends(V&& v) { return std::forward<V>(v); }
...
    explicit runner(CLI::App& app)
-      : tup(depends<Ts>(app)...) {}
+      : tup(depends<Ts>(app)...), _app(app) {}
 
    template <std::size_t I=0>
    constexpr inline int exec() {
       if constexpr (I == sizeof...(Ts)) {
-         std::cerr << "Please run one of the subcommands available. Use --help to see what is available." << std::endl;
-         return -1;
+         std::cout << _app.help() << std::endl
```
Comment (ScottBailey) [inline tools/main.cpp:34]:
```suggestion
         std::cerr << _app.help() << std::endl
```

Comment (ScottBailey) [reply tools/main.cpp:34]: `git` prints help to stdout and exits with error. We can and should do the same.

Comment (mikelik) [reply tools/main.cpp:34]: Done.
I'm returning error `1`. For negative return codes in other places I have created an issue: https://github.com/AntelopeIO/antler-proj/issues/57

**Review by ScottBailey — APPROVED** [review]

**Pull request merged**